src/main/java/edu/harvard/iq/dataverse/DatasetVersionServiceBean.java

```diff
@@ -1195,4 +1195,25 @@ private DatasetVersion getPreviousVersionWithUnf(DatasetVersion datasetVersion)
     public DatasetVersion merge( DatasetVersion ver ) {
         return em.merge(ver);
     }
+    
+    /**
+     * Execute a query to return DatasetVersion
+     * 
+     * @param queryString
+     * @return 
+     */
+    public List<DatasetVersion> getUnarchivedDatasetVersions(){
+        
+        try {
+            @SuppressWarnings("unchecked")
+            List<DatasetVersion> dsl = em.createNamedQuery("DatasetVersion.findUnarchivedReleasedVersion").getResultList();
```

Comment:
I think Named Query should still be able to take a type, si you don't need the SuppressWarnings.